robot/impl/resource_manager.go

```diff
@@ -614,6 +610,7 @@ func (manager *resourceManager) completeConfig(
 					defer func() {
 						stopSlowLogger()
 						resChan <- struct{}{}
+						lr.reconfigureWorkers.Done()
```

Comment:
I'm a little suspicious of what `reconfigureWorkers` really represents. It looks like we're `Add`ing and `Done`ing twice for `processResource` calls if `!syncRes`. That seems wrong to me. It's only used in testing from what I can tell, so I won't block this PR on it; just noting that this caught my eye 👁️ .

Reply:
since reconfiguration is done level by level, the robot starts a goroutine to manage each separate resource configuration for each level (which also starts and waits for it's own goroutine but has the escape hatch of 1min) in parallel. it probably can be managed better by StoppableWorker haha, but it does make sense

Reply:
Ah yeah if we're two levels of goroutine deep that does make sense. Forgot some of the nuances of this timeout system. `StoppableWorkers` might be good, but I'm more concerned with getting that struct into signaling for now.